editoast/src/views/infra/railjson.rs

```diff
@@ -0,0 +1,81 @@
+use crate::api_error::{ApiError, ApiResult};
+
+use crate::db_connection::DBConnection;
+use crate::infra::RAILJSON_VERSION;
+use crate::schema::RailJson;
+use diesel::sql_types::Integer;
+use diesel::sql_types::Text;
+use diesel::{sql_query, RunQueryDsl};
+use rocket::http::Status;
+use rocket::response::status::Custom;
+use rocket::serde::json::{Error as JsonError, Json};
+use thiserror::Error;
+
+#[derive(QueryableByName)]
+struct RailJsonData {
+    #[sql_type = "Text"]
+    railjson: String,
+}
+
+#[derive(Debug, Error)]
+enum ListErrorsRailjson {
+    #[error("Wrong Railjson version provided")]
+    WrongRailjsonVersionProvided,
+}
+
+impl ApiError for ListErrorsRailjson {
+    fn get_status(&self) -> Status {
+        Status::BadRequest
+    }
+
+    fn get_type(&self) -> &'static str {
+        match self {
+            ListErrorsRailjson::WrongRailjsonVersionProvided => {
+                "editoast:infra:railjson:WrongRailjsonVersionProvided"
+            }
+        }
+    }
+}
+/// Serialize an infra
+#[get("/<infra>/railjson?<exclude_extensions>")]
+pub async fn get_railjson(
+    infra: i32,
+    exclude_extensions: bool,
+    conn: DBConnection,
+) -> ApiResult<Custom<String>> {
+    if exclude_extensions {
+        let query = include_str!("sql/get_infra_no_ext.sql");
+        let railjson: RailJsonData = conn
+            .run(move |conn| sql_query(query).bind::<Integer, _>(infra).get_result(conn))
+            .await?;
+        Ok(Custom(Status::Ok, railjson.railjson))
+    } else {
+        let query = include_str!("sql/get_infra_with_ext.sql");
+        let railjson: RailJsonData = conn
+            .run(move |conn| sql_query(query).bind::<Integer, _>(infra).get_result(conn))
+            .await?;
+        Ok(Custom(Status::Ok, railjson.railjson))
+    }
+}
+
+/// Import an infra
+#[post("/railjson?<name>", data = "<data>")]
+pub async fn post_railjson(
+    name: String,
+    data: Result<Json<RailJson>, JsonError<'_>>,
+    conn: DBConnection,
+) -> ApiResult<Custom<String>> {
```

Comment:
- Can you check that the crated infra has a generated version lower than the infra version?
- How about adding a query parameter `generate_data`, a boolean indicating whether we should generate the data for the infrastructure or not?